// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

message BytracePluginConfig {
    uint32 buffe_size = 1;          // Sets the size of the buffer (KB) for storing and reading traces.
    repeated string categories = 2; // Lists available bytrace categories.
    uint32 time = 3;                // Sets the bytrace running duration in seconds (5s by default)
    string clock = 4;               // Sets the clock, which can be boot (default), global, mono, uptime, or perf.
    string outfile_name = 5;        // Sets the name of the target file (stdout by default).
    bool is_root = 6;
}